test/CMakeLists.txt

```diff
@@ -2,6 +2,7 @@ add_executable(test_detail test_detail.cpp)
 target_include_directories(test_detail PRIVATE "${PROJECT_SOURCE_DIR}/include/")
 target_link_libraries(test_detail
     nlohmann_json
+    Catch2Main
     Catch2
```

Comment:
Do we still need the Catch2 links here?

Reply:
Yes, we need both and in this exact order :) cmake is so fun...